library/htmlspecialchars.inc.php

```diff
@@ -51,6 +51,19 @@ function errorLogEscape($text)
     return attr($text);
 }
 
+/**
+ * Escape variables that are outputted into csv and spreadsheet files.
+ * See here: https://www.owasp.org/index.php/CSV_Injection
+ * Based mitigation strategy on this report: https://asecurityz.blogspot.com/2017/12/csv-injection-mitigations.html
+ *  1. Remove all the following characters:  = + - @ " |
+ *  2. Prepend value with a space.
+ *  3. Remove trailing spaces (this specific strategy was reported here: https://www.contextis.com/en/blog/comma-separated-vulnerabilities).
+ */
+function csvEscape($text)
+{
+    return " " . rtrim(preg_replace('/[=+-@"|]/', '', $text));
+}
```

Comment:
@stephenwaite ,
This is the overkill method :)
If needed, we can make another parameter with level of overkill we need, or else I am guessing this will break things on certain places. This PR will address the reported security vulnerability and I'll test it out and see if it breaks any of the csv batchcom output.

Reply:
btw, this is really something that should be protected against on the excel/spreadsheet side and not by csv creation software... :(

Reply:
ok, nice, will take a closer look at the php
yeah, saw it mentioned that while the evil formulas could be escaped on the first import might then be propagated when libre writer strips it out upon saving

Reply:
Update it a bit to be more clear and not break common stuff.

Reply:
This solution is working well on batchcom reporting :)